lib/Xmds/Soap.php

```diff
@@ -1623,6 +1641,7 @@ protected function doSubmitStats($serverKey, $hardwareKey, $statXml)
                 'widgetId' => (int) $widgetId,
                 'duration' => (int) $duration,
                 'count' => ($count != '') ? (int) $count : 1,
+                'engagements' => (count($engagements) > 0) ? json_encode($engagements) : '[]',
```

Comment:
Does this mean we are `json_encoding` and then on the other side in the case of mongo `json_decoding` again? 
It would make more sense (I think) to pass the array here, and then `json_encode` only for MySQL?